File: usacobronze/whereami.cpp
#include <string>
#include <fstream>
using namespace std;
string s;
int k;
int ct(string targ, int tlen) {
    int count=0;
    for (int letter=0;letter<k;letter++) {
        if (s.substr(letter,tlen)==targ) count++;
    }
    return count;
}
int main() {
    ifstream cin("whereami.in");
    ofstream cout("whereami.out");
    cin >> k >> s;
    for (int x=1;x<k+2;x++) {
        bool is1=true;
        for (int y=0;y<k-x;y++) {
            if (ct(s.substr(y,x),x)!=1) {
                is1=false;
                break;
            }
        }
        if (is1) {
            cout << x << endl;
            return 0;
        }
    }
    return -1;

}

File: usacobronze/whereami.py
with open("whereami.in","r") as fin:
    k,s=fin.readlines()
k,s=int(k),s[:-1]
def ct(targ:str,tlen):
    # counts how many occurrences a substring occurs in a string.
    # hey! This looks like str.count(). But it's not. For example, "aaa".count("aa") gives 1. I want it to give 2
    # because that's what I need. If a substring appears more than once, then it's not unique and we need to bump up
    # the total length by 1.
    return sum(s[letter:letter + tlen] == targ for letter in range(k))
for x in range(1,k+2): # length of the unique substring
    is1 = True
    for y in range(k-x):
        if ct(s[y:y+x],x)!=1:
            is1=False
            break
    if is1:
        break
with open("whereami.out","w") as fout:
    fout.write(str(x))
